src/actions/bridgeWriteContract.ts

```diff
@@ -0,0 +1,50 @@
+import {
+  type PublicClient,
+  Chain,
+  Transport,
+  encodeFunctionData,
+  WriteContractParameters,
+  Account,
+  Abi,
+  EncodeFunctionDataParameters,
+} from 'viem'
+import { writeContract } from 'viem/actions'
+import { l1CrossDomainMessengerABI } from '@eth-optimism/contracts-ts'
+import { OpChainL2, OpChainL1 } from '@roninjin10/rollup-chains'
+
+export async function bridgeWriteContract<
+  TAbi extends Abi | readonly unknown[] = Abi,
+  TFunctionName extends string = string,
+  TChainL1 extends OpChainL1 | undefined = OpChainL1,
```

Comment:
can't wait get this from the client?

Reply:
We are getting it from the client it is passed in as a generic to be inferred from the client